gluten-data/src/main/scala/io/glutenproject/vectorized/GlutenColumnarBatchSerializer.scala

```diff
@@ -142,6 +141,7 @@ private class GlutenColumnarBatchSerializerInstance(schema: StructType,
             readBatchNumRows.set(numRowsTotal.toDouble / numBatchesTotal)
           }
           numOutputRows += numRowsTotal
+          cSchema.close()
```

Comment:
@kerwin-zk After making this change, It seems we have memory leak in spark 3.2, and it doesn't exit in spark 3.3. I will try to find why it leads to memory leak in spark 3.2. If not find, I would revert this change and move it to another pr.

Reply:
Thanks @jackylee-ch and is there any updates about this?

Reply:
Yes, it has been fixed.